Cargo.toml

```diff
@@ -100,6 +102,9 @@ http-body-util = "0.1.2"
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 shared_memory = { workspace = true }
 
+[target.'cfg(any(windows, all(unix, not(target_os = "emscripten"), not(target_os = "ios"), not(target_os = "macos"))))'.dependencies]
```

Comment:
You could also consider something similar to what wgpu does: https://github.com/gfx-rs/wgpu/blob/trunk/wgpu/build.rs

Reply:
@wkozyra95 @WojciechBarczynski this has to be so complicated because we're planning to release this library separately, so it should work on windows, android, bsd and other vulkan-supporting targets too.

@noituri very nice idea, will do that

Reply:
ok, I was too enthusiastic. The lib wgpu uses cannot be used in Cargo.tomls. I'll add a comment explaining the directive.

Reply:
oh, just noticed that this particular directive is not actually in the library. I can change it to just work on linux then, but are we sure we want to do that?

Reply:
no, given that we plan to publish it, let's make it work everywhere where possible